File: cpp/top_k_frequent_elements.cc
class Solution {
public:
  vector<int> topKFrequent(vector<int>& nums, int k) {
    priority_queue<pair<int, int>, vector<pair<int, int>>,
                   greater<pair<int, int>>> pq;
    unordered_map<int, int> count;
    for (const auto &num : nums) {
      count[num] += 1;
    }
    for (const auto &kv : count) {
      pq.push({kv.second, kv.first});
      if (pq.size() > k) {
        pq.pop();
      }
    }
    vector<int> result;
    for (; !pq.empty(); pq.pop()) {
      result.push_back(pq.top().second);
    }
    return result;
  }
};

File: python/top_k_frequent_elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        freq = {}
        freq_list = []
        for num in nums:
            if num in freq:
                freq[num] += 1
            else:
                freq[num] = 1
        for key in freq.keys():
            freq_list.append((-freq[key], key))
        heapq.heapify(freq_list)
        topk = []
        for i in range(k):
            topk.append(heapq.heappop(freq_list)[1])
        return topk
